src/gui/guiScrollBar.cpp

```diff
@@ -183,36 +183,41 @@ void GUIScrollBar::draw()
 			slider_rect.LowerRightCorner.Y =
 					slider_rect.UpperLeftCorner.Y + thumb_size;
 		}
-		skin->draw3DButtonPaneStandard(this, slider_rect, &AbsoluteClippingRect);
+		
+		// Clip scrollbar so it doesn't show behind buttons for elastic scrolling
+		if (up_button->isVisible() || down_button->isVisible()) {
+            core::rect<s32> clip = AbsoluteClippingRect;
+            if (up_button->isVisible())
+                clip.UpperLeftCorner.Y = up_button->getAbsolutePosition().LowerRightCorner.Y;
+            if (down_button->isVisible())
+                clip.LowerRightCorner.Y = down_button->getAbsolutePosition().UpperLeftCorner.Y;
+            skin->draw3DButtonPaneStandard(this, slider_rect, &clip);
+        } else {
+            // Just clip it directly
+            skin->draw3DButtonPaneStandard(this, slider_rect, &AbsoluteClippingRect);
+        }
 	}
 	IGUIElement::draw();
 }
 
-static inline s32 interpolate_scroll(s32 from, s32 to, f32 amount)
+f32 GUIScrollBar::interpolate_scroll(f32 from, f32 to, f32 amount)
 {
-	s32 step = core::round32((to - from) * core::clamp(amount, 0.001f, 1.0f));
-	if (step == 0)
-		return to;
+	f32 step = (to - from) * (amount * (last_delta_ms / 16.667f));
 	return from + step;
 }
 
 void GUIScrollBar::interpolatePos()
 {
-	if (target_pos.has_value()) {
-		// Adjust to match 60 FPS. This also means that interpolation is
-		// effectively disabled at <= 30 FPS.
-		f32 amount = 0.5f * (last_delta_ms / 16.667f);
-		setPosRaw(interpolate_scroll(scroll_pos, *target_pos, amount));
-		if (scroll_pos == target_pos)
-			target_pos = std::nullopt;
-
-		SEvent e;
-		e.EventType = EET_GUI_EVENT;
-		e.GUIEvent.Caller = this;
-		e.GUIEvent.Element = nullptr;
-		e.GUIEvent.EventType = EGET_SCROLL_BAR_CHANGED;
-		Parent->OnEvent(e);
-	}
+    // Adjust to match 60 FPS. This also means that interpolation is
+    // effectively disabled at <= 30 FPS.
+    setPosRaw(interpolate_scroll(scroll_pos, target_pos, 0.2f));
```

Comment:
Why 0.2?

Reply:
This is a magic number in a strange place that I decided was the best "interpolation" speed.

Should I allow customization on it?